scripts/l10n.ts

```diff
@@ -201,19 +219,65 @@ function countryToFlag(isoCode: string) {
     : isoCode;
 }
 
-async function generateReport(
-  missingTranslations: Record<string, { path: string; locations: number[] }>,
-) {
+interface MissingKey {
+  currentLineContent: string;
+  lineIndex: number;
+}
+interface MissingTranslations {
+  [localeCode: string]: {
+    [packageCode: string]: {
+      path: string;
+      missingKeys: MissingKey[];
+    };
+  };
+}
+
+async function generateReport(missingTranslations: MissingTranslations) {
   const lastCommitRef = await git('log -n 1 --pretty="format:%H"');
   const lines: string[] = [];
-  Object.entries(missingTranslations).forEach(([code, info]) => {
-    if (info.locations.length === 0) {
-      return;
-    }
-    lines.push(`### ${countryToFlag(code.slice(2))} ${code.slice(0, 2)}-${code.slice(2)}`);
-    info.locations.forEach((location) => {
-      const permalink = `${SOURCE_CODE_REPO}/blob/${lastCommitRef}/${info.path}#L${location}`;
-      lines.push(permalink);
+  Object.entries(missingTranslations).forEach(([languageCode, infoPerPackage]) => {
+    lines.push('');
+    lines.push(
+      `### ${countryToFlag(languageCode.slice(2))} ${languageCode.slice(0, 2)}-${languageCode.slice(
+        2,
+      )}`,
+    );
+
+    packagesWithL10n.forEach(({ key: packageKey, reportName, localesRelativePath }) => {
+      const info = infoPerPackage[packageKey];
+
+      lines.push('<details>');
+
+      const fileName = `${languageCode.slice(0, 2).toLowerCase()}${languageCode
+        .slice(2)
+        .toUpperCase()}.ts`;
+      const filePath = `${localesRelativePath}/${fileName}`;
+      if (!info) {
+        lines.push(` <summary>${reportName}: file to create </summary>`);
+        lines.push('');
+        lines.push(` > Add file \`${filePath}\` to start contribution to this locale`);
+      } else if (info.missingKeys.length === 0) {
+        lines.push(` <summary>${reportName} (Done ✅)</summary>`);
+        lines.push('');
+        lines.push(` > This locale has been completed by the community 🚀`);
+        lines.push(
+          ` > You can still look for typo fix or improvements in [the translation file](${SOURCE_CODE_REPO}/blob/${lastCommitRef}/${filePath}) 🕵`,
+        );
+      } else {
+        lines.push(` <summary>${reportName}(${info.missingKeys.length} remaining)</summary>`);
+        lines.push('');
+        info.missingKeys.forEach((missingKey) => {
+          const permalink = `${SOURCE_CODE_REPO}/blob/${lastCommitRef}/${info.path}#L${missingKey.lineIndex}`;
+          let lineContent = missingKey.currentLineContent;
+
+          if (lineContent[lineContent.length - 1] === ',') {
+            lineContent = lineContent.slice(0, lineContent.length - 1);
+          }
+          lines.push(` - [\`${lineContent}\`](${permalink})`);
```

Comment:
We need to escape ``. The line content may contain backstick when string interpolation is used, e.g. `groupColumn`. Adding a space before closing seems to work. https://meta.stackexchange.com/a/138916

```suggestion
          lines.push(` - [\``${lineContent}\`](${permalink}) ``); // Add space before `` to escape `
```

Reply:
Nice, I did not know about this space technic. Seems to work well 🔽


### 🇺🇦 uk-UA
<details>
 <summary>🧑‍💼 DataGrid, DataGridPro, DataGridPremium(16 remaining)</summary>

 - [ `` toolbarQuickFilterPlaceholder: 'Search...' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L54)
 - [ `` toolbarQuickFilterLabel: 'Search' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L55)
 - [ `` toolbarQuickFilterDeleteIconLabel: 'Clear' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L56)
 - [ `` toolbarExportExcel: 'Download as Excel' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L63)
 - [ `` filterPanelLinkOperator: 'Logic operator' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L75)
 - [ `` filterOperatorIsAnyOf: 'is any of' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L98)
 - [ `` checkboxSelectionSelectAllRows: 'Select all rows' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L141)
 - [ `` checkboxSelectionUnselectAllRows: 'Unselect all rows' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L142)
 - [ `` checkboxSelectionSelectRow: 'Select row' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L143)
 - [ `` checkboxSelectionUnselectRow: 'Unselect row' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L144)
 - [ `` groupingColumnHeaderName: 'Group' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L164)
 - [ `` groupColumn: name => `Group by ${name}` ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L165)
 - [ `` unGroupColumn: name => `Stop grouping by ${name}` ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L166)
 - [ `` expandDetailPanel: 'Expand' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L169)
 - [ `` collapseDetailPanel: 'Collapse' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L170)
 - [ `` rowReorderingHeaderName: 'Row reordering' ``](https://github.com/mui/mui-x/blob/6840befc70c5e0bbcdf18e33cecb762d74169d86/packages/grid/x-data-grid/src/locales/ukUA.ts#L173)
</details>
<details>
 <summary>🕒 Date and Time Pickers: file to create </summary>

 > Add file `packages/x-date-pickers/src/locales/ukUA.ts` to start contribution to this locale
</details>